components/dashboard/src/consent-cookie.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import Cookies, { CookieAttributes } from "js-cookie";
+
+export const COOKIE_NAME = `${window.location.hostname}-consent`;
+
+export let cookieConsentDefaultValue = {
+    necessary: true,
+    analytical: false,
+    targeting: false,
+};
+
+export function cookieSettingOptions(domain: string): CookieAttributes {
+    return {
+        domain: `.${domain}`,
+        expires: 365,
+        path: "/",
+        secure: false,
+        sameSite: "lax",
+    };
+}
+
+export function setNonNecessaryValues(analyticalValue: boolean, targetingValue: boolean) {
```

Comment:
nit: if we had a [CookieSettings](https://github.com/gitpod-io/gitpod/pull/9299/files#r855087946) we could make the interface even nicer by taking `ConsentCookie.setValues(values: Drop<CookieSettings, "necessary">)`.